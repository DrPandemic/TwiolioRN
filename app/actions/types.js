// @flow

import { PhoneNumber, Message } from '../types';

export const SET_FETCHED_ACCOUNT_NUMBERS = 'SET_FETCHED_ACCOUNT_NUMBERS';
export const FETCH_ACCOUNT_NUMBERS = 'FETCH_ACCOUNT_NUMBERS';
export const FETCH_ACCOUNT_NUMBER_ERROR = 'FETCH_ACCOUNT_NUMBER_ERROR';

export const SELECT_NUMBER = 'SELECT_NUMBER';
export const SELECT_CONVERSATION = 'SELECT_CONVERSATION';

export const FETCH_MESSAGES = 'FETCH_MESSAGES';
export const SET_FETCHED_MESSAGES = 'SET_FETCHED_MESSAGES';
export const FETCH_MESSAGE_ERROR = 'FETCH_MESSAGE_ERROR';

export type FetchAccountNumbersT = {|
  type: 'FETCH_ACCOUNT_NUMBERS',
|}
export type SuccessFetchAccountNumbersT = {|
  type: 'SET_FETCHED_ACCOUNT_NUMBERS',
  fetchedAccountNumbers: Array<PhoneNumber>,
|}
export type FailFetchAccountNumbersT = {|
    type: 'FETCH_ACCOUNT_NUMBER_ERROR',
    error: any,
|}

export type SelectNumberT = {|
  type: 'SELECT_NUMBER',
  selectedNumber: ?string,
|}

export type FetchMessagesT = {|
  type: 'FETCH_MESSAGES',
|}
export type SuccessFetchMessagesT = {|
    type: 'SET_FETCHED_MESSAGES',
    fetchedMessages: Array<Message>,
|}
export type FailFetchMessagesT = {|
    type: 'FETCH_MESSAGE_ERROR',
    error: any,
|}

export type ActionT =
  | FetchAccountNumbersT
  | SuccessFetchAccountNumbersT
  | FailFetchAccountNumbersT
  | SelectNumberT
  | SelectConversationT
  | FetchMessagesT
  | SuccessFetchMessagesT
  | FailFetchMessagesT;
