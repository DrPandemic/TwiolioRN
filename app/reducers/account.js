import createReducer from '../lib/createReducer'
import * as types from '../actions/types'

export const fetchAccountNumbers = createReducer({}, {
  [types.SET_ACCOUNT_NUMBER] (state, action) {
    return {
      ...state,
      accountNumbers: action.accountNumbers,
    };
  },
});

