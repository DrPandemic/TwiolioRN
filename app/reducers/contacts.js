// @flow

import { loop, Cmd } from 'redux-loop';
import RNContacts from 'react-native-contacts';


import { Contact } from '../types';
import effects from '../effects';
import * as types from '../actions/types';
import createReducer from '../lib/createReducer';
import * as contactActions from '../actions/contacts';
import { persistStore } from '../actions/persist';

export type T = {|
  loading: boolean,
  contacts: Array<Contact>,
  error: any,
|};
export const initialState: T = {
  loading: false,
  contacts: [],
  error: null,
};

export const reducer = createReducer({
  [types.TICK](state: T) {
    return loop(
      { ...state },
      Cmd.action(contactActions.fetchContacts())
    );
  },
  [types.FETCH_CONTACTS](
    state: T,
  ) {
    return loop(
      { ...state, loading: true },
      Cmd.run(effects.fetchContacts, {
        successActionCreator: contactActions.successFetchContacts,
        failActionCreator: contactActions.failFetchContacts,
        args: [RNContacts],
      })
    );
  },
  [types.SUCCESS_FETCH_CONTACTS](
    state: T,
    action: types.SuccessFetchContactsT,
  ) {
    return loop({
      ...state,
      contacts: action.contacts,
      loading: false,
      error: null,
    },
    Cmd.action(persistStore())
    );
  },
  [types.FAIL_FETCH_CONTACTS](
    state: T,
    action: types.FailFetchContactsT,
  ) {
    return loop({
      ...state,
      loading: false,
      error: action.error,
    },
    );
  },
});
