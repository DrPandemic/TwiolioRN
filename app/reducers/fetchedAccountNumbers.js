// @flow

import { loop, Cmd } from 'redux-loop';
import * as types from '../actions/types';
import * as fetchActions from '../actions/fetchedAccountNumbers';
import { getApi } from '../lib/api';
import createReducer from '../lib/createReducer';
import effects from '../effects';
import { PhoneNumber } from '../types';

export type T = {|
  loading: boolean,
  numbers: Array<PhoneNumber>,
  error: any,
|};
export const initialState: T = {
  loading: false,
  numbers: [],
  error: {},
};

export const reducer = createReducer({
  [types.TICK](state: T) {
    return loop(
      { ...state },
      Cmd.action(fetchActions.fetchAccountNumbers()),
    );
  },
  [types.FETCH_ACCOUNT_NUMBERS](state: T) {
    return loop(
      { ...state, loading: true },
      Cmd.run(effects.fetchNumbers, {
        successActionCreator: fetchActions.successFetchAccountNumbers,
        failActionCreator: fetchActions.failFetchAccountNumbers,
        args: [getApi()],
      })
    );
  },
  [types.SET_FETCHED_ACCOUNT_NUMBERS](
    state: T,
    action: types.SuccessFetchAccountNumbersT
  ) {
    return {
      ...state,
      numbers: action.fetchedAccountNumbers,
      error: {},
      loading: false,
    };
  },
  [types.FETCH_ACCOUNT_NUMBER_ERROR](
    state: T,
    action: types.FailFetchAccountNumbersT
  ) {
    return {
      ...state,
      error: action.error,
      loading: false,
    };
  },
});
