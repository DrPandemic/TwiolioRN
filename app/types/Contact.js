// @flow

import {
  PhoneNumberFormat as PNF,
  PhoneNumberUtil
} from 'google-libphonenumber';

import { Message } from './';
import { flatMap } from '../lib/array';

const phoneUtil = PhoneNumberUtil.getInstance();

export type Email = {|
  label: string,
  email: string
|};

export type ContactPhoneNumber = {|
  label: string,
  number: string
|};

export type PostalAddress = {|
  postCode: string,
  city: string,
  neighborhood: string,
  street: string,
  formattedAddress: string,
  label: string
|};

export default class Contact {
  recordID: string;
  company: string;
  emailAddresses: Array<Email>;
  familyName: string;
  givenName: string;
  jobTitle: string;
  middleName: string;
  phoneNumbers: Array<ContactPhoneNumber>;
  hasThumbnail: boolean;
  thumbnailPath: string;
  postalAddresses: Array<PostalAddress>;

  constructor(response: any) {
    this.recordID = response.recordID;
    this.company = response.company;
    this.emailAddresses = response.emailAddresses;
    this.familyName = response.familyName;
    this.givenName = response.givenName;
    this.jobTitle = response.jobTitle;
    this.middleName = response.middleName;
    this.phoneNumbers = response.phoneNumbers.map(
      p => ({
        ...p,
        number: phoneUtil.format(phoneUtil.parse(p.number, 'US'), PNF.E164),
      })
    );
    this.hasThumbnail = response.hasThumbnail;
    this.thumbnailPath = response.thumbnailPath;
    this.postalAddresses = response.postalAddresses;
  }
}

export type PhoneNumberWithContact = {
  contact: Contact,
  phoneNumber: string
};

export function findContactsForNumber(
  contacts: Array<Contact>,
  number: string,
): Array<Contact> {
  const formattedNumber = phoneUtil.parse(number);
  return contacts.filter(
    c => c.phoneNumbers.filter(
      n => phoneUtil.isNumberMatch(phoneUtil.parse(n.number), formattedNumber) >= 2
    ).length > 0
  );
}

export function getNameForMessage(
  contacts: Array<Contact>,
  message: Message,
): string {
  const number = message.conversationUsers.other;
  const foundContacts = findContactsForNumber(contacts, number);

  return foundContacts.length > 0 ? foundContacts[0].givenName : number;
}

export function expandPhoneNumbers(contacts: Array<Contact>): Array<PhoneNumberWithContact> {
  return flatMap(contacts,
    contact => contact.phoneNumbers.map(
      n => ({ contact, phoneNumber: n.number })
    ));
}
