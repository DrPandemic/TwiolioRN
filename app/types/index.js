// @flow
import Message from './Message';
import PhoneNumber from './PhoneNumber';
import ConversationStore from './ConversationStore';

export {
  Message,
  PhoneNumber,
  ConversationStore,
};
