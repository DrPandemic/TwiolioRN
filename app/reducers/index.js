import { combineReducers } from 'redux';
import * as AccountReducers from './account'

export default combineReducers(Object.assign({},
  AccountReducers,
));
