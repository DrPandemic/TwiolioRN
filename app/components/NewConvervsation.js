// @flow

import React, { Component } from 'react';
import {
  Dimensions,
  FlatList,
  View,
  StyleSheet
} from 'react-native';
import { List, ListItem } from 'react-native-elements';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { ActionCreators } from '../actions';
import { Contact } from '../types';
import { getConversations, filterByUs } from '../types/ConversationStore';
import { expandPhoneNumbers } from '../types/Contact';
import { Colors } from '../constants';
import type { StateT } from '../reducers';
import type { T as ContactT } from '../reducers/contacts';
import type { PhoneNumberWithContact } from '../types/Contact';

const { width } = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    width,
    flexDirection: 'column',
    backgroundColor: Colors.background,
  },
  list: {
    marginTop: 0,
    borderBottomWidth: 0,
  },
  item: {
    paddingTop: 10,
    paddingBottom: 10,
    borderBottomWidth: 0,
  },
  flatList: {
    marginBottom: 30,
  },
});

type PropsT = {
  contacts: ContactT,
  push: (any) => void,
};

export class PNewConversation extends Component {
  props: PropsT;

  constructor(props: PropsT): void {
    super(props);

    this.onSelectContact = this.onSelectContact.bind(this);
  }

  onSelectContact(row: PhoneNumberWithContact): void {
    this.props.push({
      pathname: '/newConversationSource',
      state: row,
    });
  }

  renderRow(row: PhoneNumberWithContact) {
    return (
      <ListItem
        title={`${row.contact.name} - ${row.phoneNumber}`}
        onPress={() => this.onSelectContact(row)}
        containerStyle={styles.item}
        underlayColor={'#dedede'}
        hideChevron={true}
      />
    );
  }

  render() {
    const numbers = expandPhoneNumbers(this.props.contacts.contacts);

    return (
      <View style={styles.container}>
        <List style={styles.list}>
          <FlatList
            data={numbers}
            renderItem={({ item }) => this.renderRow(item)}
            keyExtractor={(row: PhoneNumberWithContact) => [row.contact.recordID, row.phoneNumber].join('')}
            style={styles.flatList}
          />
        </List>
      </View>
    );
  }
}

function mapStateToProps(state: StateT) {
  return {
    contacts: state.contacts,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(ActionCreators, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(PNewConversation);
