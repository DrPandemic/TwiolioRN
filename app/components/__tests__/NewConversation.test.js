// @flow

import renderer from 'react-test-renderer';

jest.mock('../../store');

import { PNewConversation } from '../NewConvervsation';
import contactFixture from '../../test_helpers/fixtures/contacts.json';
import { Contact } from '../../types';
import { expandPhoneNumbers } from '../../types/Contact';
import { initialState } from '../../reducers';

function createContacts(contacts: Array<any>): Array<Contact> {
  return contacts.map(c => new Contact(c));
}

test('renders a row', () => {
  const newConversation = new PNewConversation({
    contacts: {
      ...initialState.contacts,
      contacts: createContacts([contactFixture.simple[0]]),
    },
    push: () => {},
  });

  expect(
    renderer.create(newConversation.render())
  ).toMatchSnapshot();
});

test('renders rows', () => {
  const newConversation = new PNewConversation({
    contacts: {
      ...initialState.contacts,
      contacts: createContacts(contactFixture.simple),
    },
    push: () => {},
  });

  expect(
    renderer.create(newConversation.render())
  ).toMatchSnapshot();
});

test('renders empty rows', () => {
  const newConversation = new PNewConversation({
    contacts: {
      ...initialState.contacts,
      contacts: [],
    },
    push: () => {},
  });

  expect(
    renderer.create(newConversation.render())
  ).toMatchSnapshot();
});

test('renders a row with correct redirect', () => {
  const spy = jest.fn();
  const contacts = createContacts([contactFixture.simple[0]]);
  const contact = expandPhoneNumbers(contacts)[0];
  const newConversation = new PNewConversation({
    contacts: {
      ...initialState.contacts,
      contacts,
    },
    push: spy,
  });

  const renderedContent = newConversation.renderRow(contact);
  renderedContent.props.onPress();

  expect(spy).toBeCalledWith({
    pathname: '/newConversationSource',
    state: contact,
  });
});
