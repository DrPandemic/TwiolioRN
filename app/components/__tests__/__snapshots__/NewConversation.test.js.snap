// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a row 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "flexDirection": "column",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "marginTop": 0,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "contact": Contact {
              "company": "",
              "emailAddresses": "",
              "familyName": null,
              "givenName": "Foo",
              "hasThumbnail": false,
              "jobTitle": "",
              "middleName": null,
              "phoneNumbers": Array [
                Object {
                  "label": "mobile",
                  "number": "+15142222222",
                },
              ],
              "postalAddresses": Array [],
              "recordID": "1",
              "thumbnailPath": "",
            },
            "phoneNumber": "+15142222222",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginBottom": 30,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "borderBottomWidth": 0,
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#43484d",
                            "fontSize": 17.5,
                          },
                          Object {
                            "marginLeft": 10,
                          },
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    Foo   - +15142222222
                  </Text>
                </View>
                <View
                  style={undefined}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders empty rows 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "flexDirection": "column",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "marginTop": 0,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginBottom": 30,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders rows 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "flexDirection": "column",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "marginTop": 0,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "contact": Contact {
              "company": "",
              "emailAddresses": "",
              "familyName": null,
              "givenName": "Foo",
              "hasThumbnail": false,
              "jobTitle": "",
              "middleName": null,
              "phoneNumbers": Array [
                Object {
                  "label": "mobile",
                  "number": "+15142222222",
                },
              ],
              "postalAddresses": Array [],
              "recordID": "1",
              "thumbnailPath": "",
            },
            "phoneNumber": "+15142222222",
          },
          Object {
            "contact": Contact {
              "company": "",
              "emailAddresses": "",
              "familyName": null,
              "givenName": "Bar",
              "hasThumbnail": false,
              "jobTitle": "",
              "middleName": null,
              "phoneNumbers": Array [
                Object {
                  "label": "mobile",
                  "number": "+15143333333",
                },
              ],
              "postalAddresses": Array [],
              "recordID": "2",
              "thumbnailPath": "",
            },
            "phoneNumber": "+15143333333",
          },
          Object {
            "contact": Contact {
              "company": "",
              "emailAddresses": "",
              "familyName": null,
              "givenName": "Baz",
              "hasThumbnail": false,
              "jobTitle": "",
              "middleName": null,
              "phoneNumbers": Array [
                Object {
                  "label": "mobile",
                  "number": "+15143333333",
                },
              ],
              "postalAddresses": Array [],
              "recordID": "3",
              "thumbnailPath": "",
            },
            "phoneNumber": "+15143333333",
          },
          Object {
            "contact": Contact {
              "company": "",
              "emailAddresses": "",
              "familyName": null,
              "givenName": "Booing",
              "hasThumbnail": false,
              "jobTitle": "",
              "middleName": null,
              "phoneNumbers": Array [
                Object {
                  "label": "mobile",
                  "number": "+1231231235",
                },
              ],
              "postalAddresses": Array [],
              "recordID": "4",
              "thumbnailPath": "",
            },
            "phoneNumber": "+1231231235",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginBottom": 30,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "borderBottomWidth": 0,
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#43484d",
                            "fontSize": 17.5,
                          },
                          Object {
                            "marginLeft": 10,
                          },
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    Foo   - +15142222222
                  </Text>
                </View>
                <View
                  style={undefined}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "borderBottomWidth": 0,
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#43484d",
                            "fontSize": 17.5,
                          },
                          Object {
                            "marginLeft": 10,
                          },
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    Bar   - +15143333333
                  </Text>
                </View>
                <View
                  style={undefined}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "borderBottomWidth": 0,
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#43484d",
                            "fontSize": 17.5,
                          },
                          Object {
                            "marginLeft": 10,
                          },
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    Baz   - +15143333333
                  </Text>
                </View>
                <View
                  style={undefined}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "borderBottomWidth": 0,
                    "paddingBottom": 10,
                    "paddingTop": 10,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#43484d",
                            "fontSize": 17.5,
                          },
                          Object {
                            "marginLeft": 10,
                          },
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    Booing   - +1231231235
                  </Text>
                </View>
                <View
                  style={undefined}
                >
                  <View />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
